dotCMS/src/main/java/com/dotmarketing/business/VersionableAPIImpl.java

```diff
@@ -748,7 +750,15 @@ public Optional<ContentletVersionInfo> getContentletVersionInfo(final String ide
 	@Override
 	@CloseDBIfOpened
 	public List<ContentletVersionInfo> findContentletVersionInfos(final String identifier) throws DotDataException, DotStateException {
-	  return versionableFactory.findAllContentletVersionInfos(identifier);
+
+        Logger.debug(this, "Finding the contentlet version info, for the id: " + identifier);
+        final Identifier identifierBean = APILocator.getIdentifierAPI().find(identifier);
+        if (Objects.isNull(identifierBean) || !InodeUtils.isSet(identifierBean.getId())) {
+
+            throw new DoesNotExistException("Identifier not found for identifier: " + identifier);
```

Comment:
This message could be: `"(Version Infos not found for Identifier" + identifier);`